docs/sources/mimir/operators-guide/architecture/components/query-scheduler/index.md

```diff
@@ -50,14 +50,57 @@ To use the query-scheduler with DNS-based service discovery, configure the query
 
 > **Note:** The querier pulls queries only from the query-frontend or the query-scheduler, but not both. `-querier.frontend-address` and `-querier.scheduler-address` options are mutually exclusive, and only one option can be set.
 
-### Ring-based service discovery (experimental)
+### Ring-based service discovery
 
 To use the query-scheduler with ring-based service discovery, configure the query-schedulers to join their hash ring, and the query-frontends and queriers to discover query-scheduler instances via the ring:
 
 1. [Configure the hash ring]({{< relref "../../../configure/configure-hash-rings.md" >}}) for the query-scheduler.
 1. Set `-query-scheduler.service-discovery-mode=ring` (or its respective YAML configuration parameter) to query-scheduler, query-frontend and querier.
 1. Set the `-query-scheduler.ring.*` flags (or their respective YAML configuration parameters) to query-scheduler, query-frontend and querier.
 
+#### Migrate from DNS-based to ring-based service discovery
+
+To migrate the query-scheduler from [DNS-based service discovery](#dns-based-service-discovery) to [ring-based service discovery](#ring-based-service-discovery) you can follow these steps:
+
+1. Configure the **query-scheduler** instances to join a ring:
+   ```
+   -query-scheduler.service-discovery-mode=ring
+
+   # Configure the query-scheduler ring backend (e.g. "memberlist").
+   -query-scheduler.ring.store=<backend>
+
+   # If the configured <backend> is "memberlist", then ensure memberlist is configured for the query-scheduler.
+   -memberlist.join=<same as other Mimir components>
+
+   # If the configured <backend> is "consul" or "etcd", then ensure their backend configuration is set
+   # for the query-scheduler ring:
+   # - Consul: -query-scheduler.ring.consul.*
+   # - Ecd:    -query-scheduler.ring.etcd.*
+   ```
+1. Wait until query-scheduler instances rolled out. Then ensure the changes have been successfully applied:
+   - Open the [query-scheduler ring status]({{< relref "../../../reference-http-api/index.md#query-scheduler-ring-status" >}}) page and ensure all query-scheduler instances are registered to the ring.
+   - At this point, queriers and query-frontend are still discovering query-schedulers via DNS.
```

Comment:
```suggestion
1. Wait until the query-scheduler instances have completed rolling out.
1. Ensure the changes have been successfully applied; open the [query-scheduler ring status]({{< relref "../../../reference-http-api/index.md#query-scheduler-ring-status" >}}) page and ensure all query-scheduler instances are registered to the ring.
   At this point, queriers and query-frontend are still discovering query-schedulers via DNS.
```
I opted to combine these instructions because it looks like they both say the same thing while providing some context.